mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -186,10 +198,14 @@ pub struct Nexus {
     /// the offset in num blocks where the data partition starts
     pub data_ent_offset: u64,
     /// nbd device which the nexus is exposed through
-    pub(crate) nbd_disk: Option<Disk>,
+    pub(crate) nbd_disk: Option<NbdDisk>,
     /// the handle to be used when sharing the nexus, this allows for the bdev
     /// to be shared with vbdevs on top
     pub(crate) share_handle: Option<String>,
+    /// frontend share protocol used when the nexus is published
+    pub share_protocol: Option<ShareProtocolNexus>,
+    /// iscsi target which the nexus is exposed through
+    pub(crate) iscsi_target: Option<NexusIscsiTarget>,
```

Comment:
Currently, we have three optional members of the struct related to sharing and there are combinations that don't make a sense. For example, both, nbd_disk and iscsi_target, being Some(). Or share_protocol being None and nbd_disk or iscsi_target Some. How can we simplify this and eliminate inconsistent combinations at the struct definition level? 🤔 

A possible solution could be a new enum Disk (or Target) having three variants: Nbd, Iscsi and Nvmf. These variatns would hold a value NbdDisk, NexusIscsiTarget and later NexusNvmfTarget. That way we can easily enforce that only one target/disk is set for the nexus at any given time. And we would not need share_protocol at all as it is contained in the enum.

Reply:
I think that indeed we can make this generic over the ShareProtocol.

Reply:
Will do as suggested.

Reply:
Resolved in https://github.com/openebs/Mayastor/pull/125/commits/d2abbaa1d04637148b05dac687b3b713b529794b